Fixes Uncaught TypeError on getDevice due to default being empty

AltamashShaikh opened the pull request:
### Description:

Fixes Uncaught TypeError on getDevice due to default being empty
Fixes: https://github.com/matomo-org/matomo/issues/21572

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

AltamashShaikh committed:
Fixes Uncaught TypeError on getDevice due to default being empty

snake14 commented:
@AltamashShaikh It looks like the test case that I added needs to be updated corresponding to your change.

AltamashShaikh committed:
Fixes failing test and added a new test for default values

AltamashShaikh commented:
> @AltamashShaikh It looks like the test case that I added needs to be updated corresponding to your change.

@snake14 Yes, updated and added 1 more new case

snake14 reviewed (approved):
Looks alright to me. I don't see any errors when I view Device Detection in my local instance.

sgiehl commented on CachedEntry.php:
```diff
@@ -28,10 +28,10 @@ public function __construct(string $userAgent, $clientHints, array $values)
+        $this->client = $values['client'] ?? null;
+        $this->device = $values['device'] ?? null;
         $this->model = $values['model'] ?? '';
         $this->os = $values['os'] ?? '';
```
That one better should be null as well. 
```suggestion
        $this->os = $values['os'] ?? null;
```

AltamashShaikh replied:
done :+1:

AltamashShaikh committed:
Applied PR suggestion

tsteur reviewed (approved):
Nice, and thanks for adding the test 👍 💯
